src/main.rs

```diff
@@ -52,6 +58,13 @@ fn main() {
         // origin of the panic, including the payload passed to panic! and the source code location
         // from which the panic originated.
         error!("Firecracker {}", info);
+        if let Err(e) = stdin.lock().set_canon_mode() {
```

Comment:
I like this better since it is more rusty:
```
        error!("Firecracker {}", info);
        stdin.lock().set_canon_mode().unwrap_or_else(|e| {
            error!(
                "Failure while trying to reset stdin to canonical mode: {}",
                e
            )
        });

        METRICS.vmm.panic_count.inc();
        error!("{:?}", Backtrace::new());
        LOGGER
            .log_metrics()
            .unwrap_or_else(|e| error!("Failed to log metrics while panicking: {}", e));
```

Reply:
`If let Err(e)` is rusty enough for me :P
There are 2 reasons why I do not personally like `unwrap_or_else`:
* Maybe the syntax is more elegant but I think it is evil since it makes it very easy for reviewers to miss when the error does not get propagated. Paramount place where the error coming from `run_vmm_action` should get displayed in the `warn!`, right?
```
vmm.run_vmm_action().unwrap_or_else(|_| {
                                    warn!("Got a spurious notification from api thread");
                                });
```
`If let err` makes it more obvious that you'll actually want to use the error.
* `unwrap` and `expect` forces the error being propagated to implement `Debug`, but since we are printing the error with `{}`, it will also need to implement `Display`. IMO there is no need to have them both.